### PR TITLE
rpds-py: update to 0.10.2, fix crates' fetching

### PR DESCRIPTION
**Pull request by ognevnydemon**

### CODE REVIEW

**Comment by Biswa96** [comment]
++pkgrel

**Comment by raedrizqie** [comment]
while at it, update to v0.10.2

**Comment by ognevnydemon** [comment]
> while at it, update to v0.10.2

hm, should I close my PR then?

**Comment by ognevnydemon** [comment]
> why did you delete? you just need to update it

oops... I'm hurried a bit, sorry

**Comment by raedrizqie** [comment]
you forgot to update the sha256sums

**Comment by ognevnydemon** [comment]
I messed some things up... will this have bad effect?

**Comment by ognevnydemon** [comment]
@raedrizqie everything is fine now?

File: mingw-w64-python-rpds-py/PKGBUILD [hunk 1]
```diff
@@ -17,14 +17,18 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-python-build"
...
   rm -rf python-build-${MSYSTEM} | true
   cp -r "${_realname/-/_}-${pkgver}" "python-build-${MSYSTEM}"
 
   cd "${srcdir}/python-build-${MSYSTEM}"
-  cargo fetch --locked --target "${CARCH}-pc-windows-gnu"
+  local _target="${CARCH}-pc-windows-gnu"
+  if [[ $MINGW_PACKAGE_PREFIX == *-clang-aarch64 ]]; then
+    _target="${CARCH}-pc-windows-gnullvm"
+  fi
+  cargo fetch --locked --target ${_target}
```
Comment (Biswa96) [inline mingw-w64-python-rpds-py/PKGBUILD:31]: Is the `--target` option needed here? @mati865 tried that in this pull request https://github.com/msys2/MINGW-packages/pull/18231

Comment (mati865) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: It turned out in https://github.com/msys2/MINGW-packages/pull/18231 `cargo fetch --target x` will download correct dependencies but `cargo build` might want dependencies for other targets. So using `cargo fetch --target x` can cause `cargo build --release` to fail if **`--offline`** is added as well.

Comment (ognevnydemon) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: good catch. so there is no sense in it and I can just remove it?

Comment (raedrizqie) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: i just follow how Arch done it in their PKGBUILD: https://gitlab.archlinux.org/archlinux/packaging/packages/python-rpds-py/-/blame/main/PKGBUILD#L35

Comment (ognevnydemon) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: me too when I made [alacritty](https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-alacritty#L34) recipe

Comment (mati865) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: @raedrizqie yeah, I tried the same in https://github.com/msys2/MINGW-packages/pull/18231 for other crates but as it turns out, this either doesn't work well on Windows or doesn't work well in general.

Comment (ognevnydemon) [reply mingw-w64-python-rpds-py/PKGBUILD:31]: anyway we can left it now as is for this package

**Pull request merged**